app/models/embeds/pico_area.rb

```diff
@@ -3,39 +3,38 @@
 module Embeds
   # This is a simple wrapper/decorator for Api::Area.
   class PicoArea < SimpleDelegator
+    TYPES = Embeds::Pico::AreaType::ALL
+    FALLBACK_TYPE = Embeds::Pico::AreaType::Country
+
     def self.find_by_area_code(area_code)
       new Api::Area.find(area_code)
     end
 
     # Sorry for the dutch. PICO is in dutch...
     def area_name
-      return 'Nederland' if type == :land
+      return 'Nederland' if type == Embeds::Pico::AreaType::Country
 
       area.sub(/\A[^_]*/, '').humanize
     end
 
     # Sorry for the dutch. PICO is in dutch...
     def type
-      pattern_whitelist.each do |type, pattern|
-        return type if area.match?(pattern)
-      end
-      :land
+      TYPES.find { |type| area.match?(type.matcher) } || FALLBACK_TYPE
+    end
+
+    def select_value
+      type.get_select_value(area)
```

Comment:
If the area type is a `Res`, an `UnmatchableSelectValueError` is raised when calling `get_select_value`. Perhaps we should show a message informing the user that the current region isn't supported?